lib/js/atoms/SurveyQuestion/SurveyQuestionOpenEnded.vue

```diff
@@ -0,0 +1,134 @@
+<template>
+	<div
+		class="surveyQuestionOpenEnded"
+		:class="{ '-disabled': state === SURVEY_QUESTION_STATES.DISABLED }"
+	>
+		<ds-modal v-if="showModal" @close-modal="showModal = false">
+			<slot name="explanation" />
+			<div slot="footer">
+				<ds-button :type="BUTTON_TYPES.OUTLINED" @click.native="showModal = false">
+					OK, rozumiem
+				</ds-button>
+			</div>
+		</ds-modal>
+		<ds-card>
+			<template slot="content">
+				<div class="surveyQuestionOpenEnded__header">
+					<label :for="inputId">
+						{{ title }}
+					</label>
+					<icon-button
+						v-if="$slots.explanation"
+						class="surveyQuestionOpenEnded__explanation"
+						:color="BUTTON_COLORS.MINOR_SUPPORTING"
+						:icon="ICONS.FA_QUESTION_CIRCLE"
+						:size="ICON_SIZES.MEDIUM"
+						:touchable="false"
+						@click.native="showModal = true"
+					/>
+				</div>
+				<div class="surveyQuestionOpenEnded__content">
+					<survey-question-textarea
+						:id="inputId"
+						class="surveyQuestionOpenEnded__input"
+						:disabled="state === SURVEY_QUESTION_STATES.DISABLED"
+						:value="value"
+						:placeholder="placeholder"
+						@input="$emit('input', $event)"
+					/>
+				</div>
+			</template>
+		</ds-card>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/buttons';
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/typography';
+
+.surveyQuestionOpenEnded {
+	&__header {
+		@include headlineS();
+
+		display: flex;
+		// header without explanation iconButton has to be the same size as with iconButton
+		min-height: $icon-button-medium-size;
+		margin-bottom: $space-s;
+		justify-content: space-between;
+	}
+
+	&__explanation {
+		align-self: flex-start;
+	}
+
+	&__content {
+		display: flex;
+		flex-direction: column;
+	}
+
+	&__input {
+		margin: $space-xxs 0;
+	}
+}
+</style>
+
+<script lang="ts">
+import DsCard from '../Card';
+import DsModal from '../Modal';
+import DsButton, { BUTTON_COLORS, BUTTON_TYPES } from '../Button';
+import IconButton from '../IconButton';
+import { ICON_SIZES, ICONS } from '../Icon';
+import { SURVEY_QUESTION_STATES } from './SurveyQuestion.consts';
+import SurveyQuestionTextarea from './SurveyQuestionTextarea.vue';
+
+export default {
+	name: 'SurveyQuestionOpenEnded',
+	components: {
+		SurveyQuestionTextarea,
+		DsButton,
+		DsCard,
+		IconButton,
+		DsModal,
+	},
+	props: {
+		title: {
+			type: String,
+			required: true,
+		},
+		value: {
+			type: String,
+			required: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wpisz swoją odpowiedź',
+		},
+		state: {
+			type: String,
+			default: SURVEY_QUESTION_STATES.DEFAULT,
+			validate(state) {
+				return Object.values(SURVEY_QUESTION_STATES).includes(state);
+			},
+		},
+	},
+	data() {
+		return {
+			showModal: false,
+			inputId:
+				'survey-question' +
+				Math.random()
+					.toString(36)
+					.replace(/[^a-z]+/g, '')
+					.substr(0, 8),
```

Comment:
```
Math.random().toString(36).replace(/[^a-z]+/g, '')

'siv'
```
This can produce an empty string, I think.

Actually, I checked and it does, rarely:
```
i = 0; do { x = Math.random().toString(36).replace(/[^a-z]+/g, ''); i++; } while (x !== ''); console.log(i);

3371864
```

Maybe it's not a real problem but if we hit a bug because of it – good luck with debugging.

Reply:
Is it better now: [`1b7d45f` (#71)](https://github.com/bethinkpl/design-system/pull/71/commits/1b7d45fb13472731dae39efce72b359b330ebf9b) ?

Reply:
Yup, my browser froze when trying to find a collision 😛